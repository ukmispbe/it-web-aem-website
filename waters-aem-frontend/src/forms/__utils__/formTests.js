export const formTests = (wrapper) => {
     
    describe('Scenario Form Rendering', () => {
        const errorBoundaryForm = wrapper.find("ErrorBoundaryForm"); 
        it('Should render the ErrorBoundaryForm component', () => {          
        expect(errorBoundaryForm.length).toBe(1);
        });

        const errorBoundary = errorBoundaryForm.find("ErrorBoundary");  
        it('Should render the ErrorBoundary component', () => {                 
            expect(errorBoundary.length).toBe(1);
        });

        const Form = errorBoundary.find("Form");  
        it('Should render the Form component', () => {                 
            expect(Form.length).toBe(1);
        });

        const form = errorBoundary.find(".cmp-form");  
        it('Should render the form component', () => {                 
            expect(form.length).toBe(1);
        });

        const FormStateProvider = form.find("FormStateProvider");  
        it('Should render the FormStateProvider component', () => {                 
            expect(FormStateProvider.length).toBe(1);
        });

        const ErrorsProvider = FormStateProvider.find("ErrorsProvider");  
        it('Should render the ErrorsProvider component', () => {                 
            expect(ErrorsProvider.length).toBe(1);
        });

        it('Should render the Submit Button in a disabled state', () => {
            const button = wrapper.find('button');
            expect(button.length).toBe(1);
            expect(button.getDOMNode().disabled).toBe(true);            
        });
    });
};