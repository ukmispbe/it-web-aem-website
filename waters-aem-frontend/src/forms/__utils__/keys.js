export const keys = {
    email: ['email','Input','input'],
    firstName: ['text','Input','input'],
    lastName: ['text','Input','input'],
    company: ['text','Input','input'],
    password: ['password','Input','input'],
    confirmPassword: ['confirmPassword','Input','input'],
    communications: ['checkbox','checkbox', 'checkbox'],
    privacy: ['checkbox','checkbox', 'checkbox'],
    currentPassword: ['password','Input','input'],
    newPassword:    ['password','Input','input'],
    country:    ['dropdown', 'Dropdown', 'Memo(Select)'],
    captcha:    ['captcha', 'Captcha','captcha'],
    phone:  ['text','Input','input']
};
