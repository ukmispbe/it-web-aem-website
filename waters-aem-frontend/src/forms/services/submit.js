import scrollToY from './../../scripts/scrollTo';
import { parse } from 'query-string';
import SessionStore from '../../stores/sessionStore';
import DigitalData from '../../scripts/DigitalData';
import cookieStore from '../../stores/cookieStore';

const postData = async (url, data) => {
    const response = await fetch(url, {
        method: 'POST',
        credentials: 'include',
        headers: {
            'Content-Type': 'application/json'
        },
        body: JSON.stringify(data)
    });

    return await response;
};

export async function registrationSubmit(data) {
    delete data.confirmPassword;

    const isCaptcha = data.hasOwnProperty('captcha');
    if (isCaptcha) {
        this.url = `${this.url}?captcha=${data.captcha}`;
        delete data.captcha;
    }

    const localeLanguage = DigitalData.language;
    const localeCountry = DigitalData.country;
    if (
        (!localeLanguage && !localeCountry) ||
        DigitalData.country === DigitalData.globalExperience
    ) {
        localeLanguage = 'en';
        localeCountry = 'US';
    }
    data.localeCountry = localeCountry;
    data.localeLanguage = localeLanguage;

    const response = await postData(this.url, data);

    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {
        console.log(
            'registration complete  This needs finishing off later',
            response.json()
        );
    } else {
        this.setError(response);
        scrollToY(0);
    }
}

export async function troubleSigningInSubmit(data) {
    const isCaptcha = data.hasOwnProperty('captcha');

    if (isCaptcha) {
        this.url = `${this.url}&captcha=${data.captcha}`;
        delete data.captcha;
    }

    this.url = this.url.replace('{email}', data.email);
    const response = await postData(this.url, data);

    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {
        if (this.redirect) {
            window.location.href = this.redirect;
        }
    } else {
        this.setError(response);
        scrollToY(0);
    }
}

export async function resetPasswordSubmit(data) {
    const queryString = parse(window.location.search);
    const resetToken = queryString.token;
    const email = queryString.email;
    const newPassword = data.password;

    let updateMode = "reset";

    if (typeof resetToken === "undefined") {
        updateMode = "update";
    }

    let body = {
        email,
        resetToken,
        newPassword
    };

    // Remove resetToken if undefined
    if (updateMode === "update") {
        body = {
            email,
            newPassword
        };
    }

    const response = await postData(this.url, body);

    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {

        if (this.redirect) {
            window.location.replace(this.redirect);
        }

    } else {
        this.setError(response);
        scrollToY(0);
    }
}

export async function changePasswordSubmit(data) {
    delete data.confirmNewPassword;

    const queryString = parse(window.location.search);
    const email = queryString.email;
    data.email = email;

    const response = await postData(this.url, data);

    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {
        console.log(
            'update password complete.  This needs finishing off later'
        );

        if (this.callback && typeof this.callback === 'function') {
            this.callback(await response.json());
        }
    } else {
        this.setError(response);
        scrollToY(0);
    }
}

export async function personalSubmit(data) {
    const response = await postData(this.url, data);

    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {
        const submitResponse = await response.json();
        const store = new SessionStore();
        store.setUserDetails(submitResponse);
        this.setProfileData(submitResponse);

        this.callback();
    } else {
        this.setError(response);
        scrollToY(0);
    }
}

export async function signInSubmit(data) {
    const isCaptcha = data.hasOwnProperty('captcha');
    if (isCaptcha) {
        this.url = `${this.url}?captcha=${data.captcha}`;
        delete data.captcha;
    }

    const response = await postData(this.url, data);
    const responseBody = await response.json();
    // remove all previous server error notifications
    this.setError();

    if (response.status === 200) {

        let data = await response.json();

        if(data.migrated !== "Y") {
            window.location.replace(this.passwordUpdateUrl + `?email=${data.email}`);
        }

        // Temporary cookie
        document.cookie = 'WatersLoginCookie=1; path=/; domain=.waters.com';
        const signInRedirect = window.sessionStorage.getItem('signInRedirect');
        if (signInRedirect || this.redirect) {
            window.location.replace(
                signInRedirect ? signInRedirect : this.redirect
            );
        }
    } else {
        this.updateFailedAttempts('signin');
        this.setError(responseBody);
        scrollToY(0);
    }
}
