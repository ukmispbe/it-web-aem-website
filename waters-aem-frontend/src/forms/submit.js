export const registrationSubmit = data => {
    console.log(data);
};
