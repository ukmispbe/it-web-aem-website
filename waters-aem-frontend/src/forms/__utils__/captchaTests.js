import { keys} from './keys';
import { act } from "react-dom/test-utils";

export const captchaTests = (inputs, inputType) => {

    describe(`Scenario ${inputType} Input Rendering`, () => {

        const fieldValidationDisplay = inputs.find(`Memo(FieldValidationDisplay)[name="${inputType}"]`);
        it('Should render the FieldValidationDisplay field', () => {                 
            expect(fieldValidationDisplay.length).toBe(1);
        });

        const mainDiv = fieldValidationDisplay.find(`.cmp-form-field-${keys[inputType][0]}`);  
        it('Should render the main div field', () => {                 
            expect(mainDiv.length).toBe(1);
        });

        const memoInput = mainDiv.find(`Memo(${keys[inputType][1]})[name="${inputType}"]`);
        it('Should render the input wrapper field', () => {                 
            expect(memoInput.length).toBe(1);
        });

        const  asyncScriptLoaderReCAPTCHA = mainDiv.find('AsyncScriptLoader(ReCAPTCHA)');
        it('Should render the asyncScriptLoaderReCAPTCHA field', () => {                 
            expect(memoInput.length).toBe(1);
        });
        const asyncScriptLoader = asyncScriptLoaderReCAPTCHA.find('AsyncScriptLoader');
        it('Should render the asyncScriptLoader field', () => {                 
            expect(memoInput.length).toBe(1);
        });
        const reCAPTCHA = asyncScriptLoader.find('ReCAPTCHA');
        it('Should render the reCAPTCHA field', () => {                 
            expect(reCAPTCHA.length).toBe(1);
        });
        
        
        
    });
};
