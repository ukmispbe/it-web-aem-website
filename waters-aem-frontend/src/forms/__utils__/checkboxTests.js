import { keys } from './keys';

export const checkboxTests = (inputs, inputType) => {
    describe(`Scenario ${inputType} Input Rendering`, () => {
        const fieldValidationDisplay = inputs.find(`Memo(FieldValidationDisplay)[name="${inputType}"]`);
        it('Should render the FieldValidationDisplay field', () => {                 
            expect(fieldValidationDisplay.length).toBe(1);
        });

        const checkboxOrRadio = fieldValidationDisplay.find(`Memo(CheckboxOrRadio)[name="${inputType}"]`);
        it('Should render the checkboxOrRadio field', () => {                 
            expect(checkboxOrRadio.length).toBe(1);
        });

        const checkbox = checkboxOrRadio.find(`input[name="${inputType}"]`);
        it('Should render the checkbox field', () => {                 
            expect(checkbox.length).toBe(1);
        });

        const label = checkboxOrRadio.find(`label[htmlFor="${inputType}"]`);
        it('Should render the label field', () => {                 
            expect(label.length).toBe(1);
        });
    });
}