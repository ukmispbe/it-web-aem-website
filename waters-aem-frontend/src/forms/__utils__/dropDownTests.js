import { keys, keys2 } from './keys';

export const dropDownTests = (input, isReadOnly) => {

    const stateManager = input.find('StateManager');
    it('Should render the State Manager field', () => {
        expect(stateManager.length).toBe(1);
    });

    const select = input.find('Select[name="country"]');
    it('Should render the Select field', () => {
        expect(select.length).toBe(1);
    });

    const selectContainer = select.find('SelectContainer');
    it('Should render the Select Container', () => {
        expect(selectContainer.length).toBe(1);
    });
 
    const dropDownControl = selectContainer.find('.cmp-custom-dropdown__control');
    it('Should render the dropDownControl', () => {
        expect(dropDownControl.length).toBe(1);
    });

    const dropDownValueContainer = dropDownControl.find('.cmp-custom-dropdown__value-container');
    it('Should render the dropDownValueContainer', () => {
        expect(dropDownValueContainer.length).toBe(1);
    });

    const dropDownValuePlaceholder = dropDownValueContainer.find('.cmp-custom-dropdown__placeholder');
    it('Should render the dropDownValuePlaceholder', () => {
        expect(dropDownValuePlaceholder.length).toBe(1);
    });

    const inputControl = dropDownControl.find('input');
    it('Should render the dropDownValueContainer input control', () => {
        expect(inputControl.length).toBe(1);
    });

    const autosizeInput = dropDownControl.find('AutosizeInput');
    it('Should render the autosizeInput', () => {
        expect(autosizeInput.length).toBe(1);
    });

    const customDropDownInput = autosizeInput.find('div[className="cmp-custom-dropdown__input"]');
    it('Should render the customDropDownInput', () => {
        expect(customDropDownInput.length).toBe(1);
    });

    const dropDownIndicators = selectContainer.find('.cmp-custom-dropdown__indicators');
    it('Should render the dropDownIndicators', () => {
        expect(dropDownIndicators.length).toBe(1);
    });

    const dropDownIndicatorSeperator = selectContainer.find('.cmp-custom-dropdown__indicator-separator');
    it('Should render the dropDownIndicatorSeperator', () => {
        expect(dropDownIndicatorSeperator.length).toBe(1);
    });

    const dropDownIndicator = dropDownIndicators.find('.cmp-custom-dropdown__dropdown-indicator');
    it('Should render the dropDownIndicator', () => {
        expect(dropDownIndicator.length).toBe(1);
    });

    if (!isReadOnly) {
        const hiddenInputField = stateManager.find('input[type="hidden"]');
        it('Should render the hiddenInputField field', () => {
            expect(hiddenInputField.length).toBe(1);
        });
    }
}