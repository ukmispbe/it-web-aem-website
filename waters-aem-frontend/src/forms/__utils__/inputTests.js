import { passwordTests } from './passwordTests';
import { keys} from './keys';
import { dropDownTests } from './dropDownTests';
import { act } from "react-dom/test-utils";
import { validIconsTest } from '../__utils__/validIconsTest';

export const inputTests = (inputs, inputType, checkDescription, isReadOnly) => {

    describe(`Scenario ${inputType} Input Rendering`, () => {

        const fieldValidationDisplay = inputs.find(`Memo(FieldValidationDisplay)[name="${inputType}"]`);
        it('Should render the FieldValidationDisplay field', () => {                 
            expect(fieldValidationDisplay.length).toBe(1);
        });

        const mainDiv = fieldValidationDisplay.find(`.cmp-form-field-${keys[inputType][0]}`);  
        it('Should render the main div field', () => {                 
            expect(mainDiv.length).toBe(1);
        });

        const memoInput = mainDiv.find(`Memo(${keys[inputType][1]})[name="${inputType}"]`);
        it('Should render the input wrapper field', () => {                 
            expect(memoInput.length).toBe(1);
        });

        const label = memoInput.find(`label[htmlFor="${inputType}"]`);
        it('Should render the label field', () => {                 
            expect(label.length).toBe(1);
        });

        const input = memoInput.find(`${keys[inputType][2]}[name="${inputType}"]`);
        it('Should render the input field', () => {                 
            expect(input.length).toBe(1);
        });

        if (inputType === "email") {
            // console.log(input.debug({ verbose: true }));
            // console.log(fieldValidationDisplay.debug({ verbose: true }));;
            act(() => {
                input.simulate('change', { target: { value: 'paul_beard' } });
            });
            act(() => {
                input.simulate('blur');
            });
        }
        
        let isCurrentPassword = false;
        let isNewPassword = false;
        switch (inputType) {
            case "country": 
                dropDownTests(memoInput, isReadOnly);
                const iconHolderX = memoInput.find('Memo(Icons)');
                validIconsTest(memoInput, 1, isReadOnly);
                // if (!isReadOnly) {
                //     let iconHolder = memoInput.find('Memo(Icons)');
                //     const validIcon = iconHolder.find('ReactSVG[className="valid-icon"]');
                //     it('Should render the valid icon', () => {                 
                //         expect(validIcon.length).toBe(1);
                //     });
            
                //     const invalidIcon = iconHolder.find('ReactSVG[className="invalid-icon"]');
                //     it('Should render the invalid icon', () => {                 
                //         expect(invalidIcon.length).toBe(1);
                //     });
                // }
                // else {
                //     let iconHolder = memoInput.find('Memo(Icons)');
                //     const lockIcon = iconHolder.find('ReactSVG[className="lock-icon"]');
                //     console.log(iconHolder.debug());
                //     it('Should render the lock icon', () => {                 
                //         expect(lockIcon.length).toBe(1);
                //     });                   
                // }
            break;
            case "email": 
                validIconsTest(memoInput, 1, isReadOnly);
            break;
            case "password": 
                passwordTests(memoInput, isCurrentPassword, isNewPassword);
            break;
            case "newPassword": 
                isNewPassword = true;
                passwordTests(memoInput, isCurrentPassword, isNewPassword);
            break;            
            case "currentPassword": 
            isCurrentPassword = true;
                passwordTests(memoInput, isCurrentPassword, isNewPassword);
            break;
            default:
                validIconsTest(memoInput, 1, isReadOnly);
        }

        const displayMessage = memoInput.find(`Memo(DisplayMessage)[name="${inputType}"]`);
        it('Should render the displayMessage field', () => {                 
            expect(displayMessage.length).toBe(1);
        });
               
        const span = displayMessage.find('.cmp-form-field--errorText');
        it('Should render the displayMessage span', () => {                 
            expect(span.length).toBe(1);
        });

        switch (inputType) {
            case "email": 
                if (checkDescription) {
                    const desc = memoInput.find('.cmp-form_description');
                    it('Should render the Description field', () => {                 
                        expect(desc.length).toBe(1);
                    });
                }
            break;
            default:
        }
        
        
    });

   
};