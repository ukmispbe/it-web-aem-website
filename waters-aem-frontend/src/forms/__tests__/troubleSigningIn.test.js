import React from 'react';
import { mount } from 'enzyme';
import { act } from "react-dom/test-utils";
import Form from '../form'
import { registrationSubmit } from '../services/submit';
import { troubleSigningInConfig } from '../__mocks__/index';
import DigitalData from '../../scripts/DigitalData';
import { formTests } from '../__utils__/formTests';
import { inputTests } from '../__utils__/inputTests';
import  { captchaTests } from '../__utils__/captchaTests';

describe('Feature Trouble Signing In [Form]', () => {

    describe('Scenario Rendering', () => {

        let wrapper;
        act(() => {
            wrapper  = mount( <Form config={troubleSigningInConfig} submitFn={registrationSubmit} isocode={DigitalData.language} />);
        });

        // Check all the Forn & Wrapper Components
        formTests(wrapper);

        const inputs = wrapper.find("Memo(FieldValidationDisplay)");  
        it('Should render the 2 Input components', () => {                 
            expect(inputs.length).toBe(2);
        });
        
        const checkDescription = false;
        const isReadOnly = false;
        inputTests(inputs, "email", checkDescription, isReadOnly);
        captchaTests(inputs, "captcha");

        afterEach(() => {
            // const fieldValidationDisplay2 = inputs.find(`Memo(FieldValidationDisplay)[name="email"]`);
            // console.log(fieldValidationDisplay2.debug({ verbose: true }));;
        });
    });

 })