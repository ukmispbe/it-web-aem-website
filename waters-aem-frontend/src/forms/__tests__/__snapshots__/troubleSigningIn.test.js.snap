// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature:Trouble Signing In Form Scenario: Rendering When initial render Then the snapshot should match 1`] = `
<form
  autocomplete="off"
  className="cmp-form cmp-form--registration "
  data-locator="form-component"
  onSubmit={[Function]}
>
  <div
    className="cmp-form-field cmp-form-field-email"
  >
    <label
      className=""
      data-locator="email-address"
      htmlFor="email"
    >
      Email Address
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form_description"
    >
      This will be your username.
    </div>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="email"
        aria-labelledby="email"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="email"
        id="email"
        name="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="email"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      data-locator="first-name"
      htmlFor="firstName"
    >
      First Name
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="firstName"
        aria-labelledby="firstName"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="firstname"
        id="firstName"
        name="firstName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      data-locator="last-name"
      htmlFor="lastName"
    >
      Last Name
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="lastName"
        aria-labelledby="lastName"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="lastname"
        id="lastName"
        name="lastName"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-text"
  >
    <label
      className=""
      data-locator="company"
      htmlFor="company"
    >
      Company
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="company"
        aria-labelledby="company"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="company"
        id="company"
        name="company"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="text"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-dropdown"
  >
    <div
      className="cmp-form-field-dropdown--wrapper"
    >
      <label
        data-locator="country-label"
        htmlFor="country"
      >
        Country*
      </label>
      <div
        aria-describedby="cmp-custom-dropdown__single-value"
        className="cmp-form-field-dropdown--wrapper"
        data-locator="country"
        tabindex="0"
      >
        <div
          className="css-1pcexqc-container"
          id="country"
          onKeyDown={[Function]}
        >
          <div
            className="css-xov0tq-control cmp-custom-dropdown__control"
            onMouseDown={[Function]}
            onTouchEnd={[Function]}
          >
            <div
              className="css-1hwfws3 cmp-custom-dropdown__value-container cmp-custom-dropdown__value-container--has-value"
            >
              <div
                className="css-0 cmp-custom-dropdown__single-value"
              >
                United States
              </div>
              <div
                className="css-1g6gooi"
              >
                <div
                  className="cmp-custom-dropdown__input"
                  style={
                    Object {
                      "display": "inline-block",
                    }
                  }
                >
                  <input
                    aria-autocomplete="list"
                    aria-label="country"
                    autoCapitalize="none"
                    autoComplete="off"
                    autoCorrect="off"
                    disabled={false}
                    id="react-select-3-input"
                    onBlur={[Function]}
                    onChange={[Function]}
                    onFocus={[Function]}
                    spellCheck="false"
                    style={
                      Object {
                        "background": 0,
                        "border": 0,
                        "boxSizing": "content-box",
                        "color": "inherit",
                        "fontSize": "inherit",
                        "label": "input",
                        "opacity": 1,
                        "outline": 0,
                        "padding": 0,
                        "width": "1px",
                      }
                    }
                    tabIndex="-1"
                    type="text"
                    value=""
                  />
                  <div
                    style={
                      Object {
                        "height": 0,
                        "left": 0,
                        "overflow": "scroll",
                        "position": "absolute",
                        "top": 0,
                        "visibility": "hidden",
                        "whiteSpace": "pre",
                      }
                    }
                  >
                    
                  </div>
                </div>
              </div>
            </div>
            <div
              className="css-1wy0on6 cmp-custom-dropdown__indicators"
            >
              <span
                className="css-1hyfx7x cmp-custom-dropdown__indicator-separator"
              />
              <div
                aria-hidden="true"
                className="css-16pqwjk-indicatorContainer cmp-custom-dropdown__indicator cmp-custom-dropdown__dropdown-indicator"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div />
              </div>
            </div>
          </div>
          <input
            name="country"
            type="hidden"
            value="us"
          />
        </div>
        <div
          className="cmp-form-field--icons"
          data-locator="cmp-form-field-icons"
        >
          <div
            className="valid-icon"
          />
          <div
            className="invalid-icon"
          />
        </div>
      </div>
      <span
        className="cmp-form-field--errorText"
        data-locator="cmp-form-field-errorText"
      >
        
      </span>
    </div>
  </div>
  <div
    className="cmp-form-field cmp-form-field-password"
  >
    <label
      className=""
      data-locator="create-password"
      htmlFor="password"
    >
      Create Password
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="password"
        aria-labelledby="password"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="password"
        id="password"
        name="password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="password"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="showHide-icon toggled"
          onMouseDown={[Function]}
        />
        <div
          className="showHideOff-icon"
          onMouseDown={[Function]}
        />
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
    <div
      className="cmp-form-field--input-requirements"
    >
      <div
        className="requirements-title"
      >
        Your password must include
      </div>
      <div>
        <div
          className="requirements-info-svg"
          id="shortPassword"
        />
        <div
          className="requirements-info"
        >
          at least 8 characters
        </div>
      </div>
      <div>
        <div
          className="requirements-info-svg"
          id="noUppercase"
        />
        <div
          className="requirements-info"
        >
          at least 1 uppercase letter
        </div>
      </div>
      <div>
        <div
          className="requirements-info-svg"
          id="noLowercase"
        />
        <div
          className="requirements-info"
        >
          at least 1 lowercase letter
        </div>
      </div>
      <div>
        <div
          className="requirements-info-svg"
          id="noDigits"
        />
        <div
          className="requirements-info"
        >
          at least 1 number
        </div>
      </div>
      <div>
        <div
          className="requirements-info-svg"
          id="noSpecial"
        />
        <div
          className="requirements-info"
        >
          at least 1 symbol (for example, !, $, #, %)
        </div>
      </div>
    </div>
    <label
      className="cmp-form-field--label-matching"
      data-locator="confirm-password"
      htmlFor="confirmPassword"
    >
      Confirm Password
      <span
        className="cmp-form-field--required"
      >
        *
      </span>
    </label>
    
    <div
      className="cmp-form-field--input"
    >
      <input
        aria-label="confirmPassword"
        aria-labelledby="confirmPassword"
        aria-required={true}
        autocomplete="off"
        className=""
        data-locator="confirmpassword"
        id="confirmPassword"
        name="confirmPassword"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        placeholder=" "
        type="password"
      />
      <div
        className="cmp-form-field--icons"
        data-locator="cmp-form-field-icons"
      >
        <div
          className="showHide-icon toggled"
          onMouseDown={[Function]}
        />
        <div
          className="showHideOff-icon"
          onMouseDown={[Function]}
        />
        <div
          className="valid-icon"
        />
        <div
          className="invalid-icon"
        />
      </div>
    </div>
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-checkbox"
  >
    <input
      aria-disabled={false}
      aria-labelledby="communications"
      aria-required={false}
      className="valid"
      data-locator="communications"
      id="communications"
      name="communications"
      readOnly={true}
      role="checkbox"
      type="checkbox"
    />
    <a
      className="checkbox  valid"
      id="communications_link"
      onClick={[Function]}
    >
      <div />
    </a>
    <div
      className="cmp-form-field-checkbox--wrapper"
      data-locator="communications-checkbox-wrapper"
    >
      <label
        htmlFor="communications"
        onClick={[Function]}
      >
        I agree to receive marketing, product, training, support, and related communications via mail, email, or other electronic means. 
        <span
          className="cmp-form-field--optional"
        >
          (optional)
        </span>
      </label>
      
    </div>
  </div>
  <div
    className="cmp-form-field cmp-form-field-captcha"
  >
    <div
      data-locator="captcha"
    />
    <span
      className="cmp-form-field--errorText"
      data-locator="cmp-form-field-errorText"
    >
      
    </span>
  </div>
  <div
    className="cmp-form-field cmp-form-field-textwithlinks"
  >
    <div
      className="cmp-form-field-textwithlinks--privacy "
      data-locator="cmp-form-field-textwithlinks-privacy"
    >
      By clicking the CREATE ACCOUNT button, you are agreeing to our
       
      <a
        className=""
        data-locator="terms-of-use"
        href=""
        id="text-with-link-1"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        Terms of Use
      </a>
       
      and understand that the information you have provided on this website will be used and processed by Waters in accordance with the
       
      <a
        className=""
        data-locator="waters-privacy-notice."
        href=""
        id="text-with-link-3"
        rel="noopener noreferrer"
        target="_blank"
        title=""
      >
        Waters Privacy Notice.
      </a>
       
    </div>
  </div>
  <button
    aria-label="form submit button"
    className="cmp-button cmp-button--no-border cmp-form--submit"
    data-locator="form-submit"
    type="submit"
  >
    Create Account
  </button>
</form>
`;
